src/Filesystem/Attribute/UploadedFile.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the zenstruck/filesystem package.
+ *
+ * (c) Kevin Bond <kevinbond@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Zenstruck\Filesystem\Attribute;
+
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Zenstruck\Filesystem\Node\Path\Expression;
+use Zenstruck\Filesystem\Node\Path\Namer;
+
+/**
+ * @author Jakub Caban <kuba.iluvatar@gmail.com>
+ */
+#[\Attribute(\Attribute::TARGET_PARAMETER|\Attribute::TARGET_PROPERTY)]
+final class UploadedFile extends PendingUploadedFile
+{
+    public string|Namer $namer;
+
+    public function __construct(
+        public string $filesystem,
+        string|Namer|null $namer = null,
+        ?string $path = null,
+        ?array $constraints = null,
+        int $errorStatus = 422,
+        ?bool $image = null,
+    ) {
+        parent::__construct($path, $constraints, $errorStatus, $image);
+
+        $this->namer = $namer ?? new Expression('{checksum}/{name}{ext}');
```

Comment:
```suggestion
        $this->namer = $namer ?? new Expression('{checksum}-{name}{ext}');
```

To avoid tons of extra sub-directories by default.

Reply:
Hmm, yeah, that's a good point. We do have the `zenstruck:filesystem:purge` which has an option to remove empty directories. (this could be scheduled to run daily or whatever).

Let's leave as is.